### ISSUE TITLE
exiting subscription menu always exits app

### ISSUE THREAD

**Issue by lucasnz** [lucasasselli/garmin-podcasts#24]
When I push back to exit the subscription menu, it always closes the app. This seems to relate to lines 68 and 69 of MainMenu.mc

hasProgress always seems to be true but I don't see when it is set.

**Comment by lucasnz** [comment]
While I'm thinking about improvements to the subscription screen, it would be good to have visual confirmation that you have selected (or unselected) a subscription. Maybe a tick or something? What are your thoughts, I'd happily contribute a change to this.

In terms of the other item, I don't want to touch it, as I don't know what impact ignoring hasProgress will have...

**Comment by lucasasselli** [comment]
hasProgress is true when the progressbar is shown on screen (i.e. when you do a remote request). It's always true when using gPodder as a provider.

**Comment by lucasasselli** [comment]
Oh I found a typo in the ProviderWrapper! Pushed a fix for the issue. It should work as expected now! 

As for adding the confirmation: that's actually something that the app originally had, but I removed it because after adding episode management on watch it was broken and I was feeling a little bit lazy!  😄

**Comment by lucasnz** [comment]
This appears to have fixed the issue with exiting the subscription screen - so we can close that part.

I'm happy to add some kind of prompt/display that a subscription has been selected. Should it be a checkbox (like the download screen), Ui.CompactAlert, Ui.CompactPrompt, or something else?

**Comment by lucasasselli** [comment]
Maybe Ui.CompactPrompt? A "Do you want to Subscribe/Unsubscribe?" "Yes/No?"

**Comment by lucasnz** [comment]
I've started working on logic that aligns the look and feel of the subscription screens with the episode manager screens. This puts the removed episodes into a toDelete array. Seems to be working correctly (and seemed to be logical since it matched another part of the app). I'll push it through as a PR once I've finished testing.

One comment/question, the exitView() function seems to popView - which results in the app returning to the main menu after episode add/delete. Is this desired behaviour - it doesn't seem quite right to me.
